quiche/src/recovery/gcongestion/recovery.rs

```diff
@@ -368,15 +385,33 @@ impl LossThreshold {
     }
 
     fn time_thresh(&self) -> f64 {
-        self.time_thresh
+        match self.time_thresh_overhead {
+            Some(time_thresh_overhead) => 1.0 + time_thresh_overhead,
+            None => self.time_thresh,
+        }
     }
 
     fn on_spurious_loss(&mut self, new_pkt_thresh: u64) {
-        self.pkt_thresh = self
-            .pkt_thresh
-            .max(new_pkt_thresh.min(MAX_PACKET_THRESHOLD));
-
-        self.time_thresh = PACKET_REORDER_TIME_THRESHOLD;
+        match &mut self.time_thresh_overhead {
+            Some(time_thresh_overhead) => {
+                if self.pkt_thresh != DISABLE_PKT_THRESHOLD {
+                    // Disable packet threshold on first spurious loss.
+                    self.pkt_thresh = DISABLE_PKT_THRESHOLD;
+                } else {
+                    // Double time threshold but cap it at `1.0`, which ends up
+                    // being 2x the RTT.
+                    *time_thresh_overhead *= 2.0;
+                    *time_thresh_overhead = time_thresh_overhead.min(1.0);
```

Comment:
Consider adding:
  `self.time_thresh = 1.0 + *time_thresh_overhead;`

And simplifying `fn time_thresh(&self) -> f64` to just return `self.time_thresh`

Reply:
Good suggestion! Addressed in https://github.com/cloudflare/quiche/pull/2149/commits/56ae6ff02ca7ba4c3b059d2dd5b330e5e77cda46